test/db2.connection.test.js

```diff
@@ -50,3 +50,48 @@ function generateDSN(config) {
     ';CurrentSchema=' + config.schema;
   return dsn;
 }
+
+describe('lazyConnect', function() {
+  it('should skip connect phase (lazyConnect = true)', function(done) {
+    var dsConfig = {
+      host: 'invalid-hostname',
+      port: 80,
+      database: 'invalid-database',
+      username: 'invalid-username',
+      password: 'invalid-password',
+      lazyConnect: true,
+    };
+    var ds = getDS(dsConfig);
+
+    var errTimeout = setTimeout(function() {
+      done();
+    }, 2000);
+    ds.on('error', function(err) {
+      clearTimeout(errTimeout);
+      done(err);
+    });
+  });
+
+  it('should report connection error (lazyConnect = false)', function(done) {
+    var dsConfig = {
+      host: 'invalid-hostname',
+      port: 80,
+      database: 'invalid-database',
+      username: 'invalid-username',
+      password: 'invalid-password',
+      lazyConnect: false,
+    };
+    var ds = getDS(dsConfig);
+
+    ds.on('error', function(err) {
+      err.message.should.containEql('[IBM][CLI Driver]');
+      done();
+      process.exit(0);
```

Comment:
Does the addition of process.exit(0) now mean that this has to be the last test case in the file.  In other words if a test were added after this one would it be skipped?  Also, is it possible to use console.trace() to get a stack to see why it is called twice?